core/src/OAuth2/MsalTokenResponse.cs

```diff
@@ -68,11 +71,28 @@ internal class MsalTokenResponse : OAuth2ResponseBase
         public string IdToken { get; set; }
 
         [DataMember(Name = TokenResponseClaim.ExpiresIn, IsRequired = false)]
-        public long ExpiresIn { get; set; }
+        public long ExpiresIn
+        {
+            get => _expiresIn;
+            set
+            {
+                _expiresIn = value;
+                AccessTokenExpiresOn = DateTime.UtcNow + TimeSpan.FromSeconds(_expiresIn);
+            }
+        }
 
-        public DateTimeOffset AccessTokenExpiresOn
+        [DataMember(Name = TokenResponseClaim.ExtendedExpiresIn, IsRequired = false)]
+        public long ExtendedExpiresIn
         {
-            get { return DateTime.UtcNow + TimeSpan.FromSeconds(ExpiresIn); }
+            get => _extendedExpiresIn;
+            set
+            {
+                _extendedExpiresIn = value;
+                AccessTokenExtendedExpiresOn = DateTime.UtcNow + TimeSpan.FromSeconds(_extendedExpiresIn);
+            }
         }
+
+        public DateTimeOffset AccessTokenExpiresOn { get; private set; }
+        public DateTimeOffset AccessTokenExtendedExpiresOn { get; private set; }
```

Comment:
shouldn't these by in the AT class?

Reply:
The previous code had the getter to always calc based on ExtendedExpiresIn plus DateTime.UtcNow which means that as long as you hold it in memory it will never expire.  So we read from AccessTokenExpiresOn / AccessTokenExtendedExpiresOn here so that the datetime doesn't keep moving forward.